### ISSUE TITLE
Consul-template config fails with `file mapstructure decode failed`

### ISSUE THREAD

**Issue by lpaulmp** [fpco/fpco-salt-formula#154]
Consul-templeta fails to initialize the service when retry and token are uncommented.
https://github.com/fpco/fpco-salt-formula/blob/94b3c899b58d0d76197f09c6fec51661976ddc3d/consul/template-tool/files/base_config.hcl#L14-L15
These produce the following error when executes the consult-template formula.
`salt-call --local state.sls consul.template-tool` .

```
[ERR] (cli) from file: /home/consul/consul-template/00-consul-template.hcl: mapstructure decode failed: 1 error(s) decoding:
* '' has invalid keys: retry, token
```
This is a [similar](https://github.com/claranet/puppet-consul_template/issues/103) issue but in puppet_consul-template

**Comment by lpaulmp** [comment]
I didn't try yet with the last version of consult-template but I tested adding the environment variable `CONSUL_TOKEN` and it worked but it wasn't the same story for retry, it didn't work adding the same config that in the [configuration-file-format
](https://github.com/hashicorp/consul-template#configuration-file-format) specifies.

```hcl
retry {
    attempts = 12
    backoff = "250ms"
    max_backoff = "1m"
}
```
Adding the configuration for `wait` worked, I started to to think that there is some bug here [retry](https://github.com/hashicorp/consul-template/blob/master/config/retry.go#L24) setup
```hcl
wait {
  min = "5s"
  max = "10s"
}
```

**Comment by ketzacoatl** [comment]
OK, sounds like you have found/confirmed a working config and we're good to update the formula with that config?

**Comment by lpaulmp** [comment]
Yes, added the environment variable `CONSUL_TOKEN` fixes the issue but we have to keep` retry` commented until finding a solution.  Probably in the last version of consult-template the issue disappears.

I'll push a fix for  the token late today
